flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTracker.java

```diff
@@ -257,18 +258,19 @@ private Map<TaskManagerLocation, ImmutableSet<ExecutionAttemptID>> groupExecutio
                         executionVertex.getExecutionState());
                 continue;
             }
-            TaskManagerLocation tmLocation = executionVertex.getCurrentAssignedResourceLocation();
-            if (tmLocation == null) {
-                LOG.trace("ExecutionVertex {} is currently not assigned", executionVertex);
-                continue;
-            }
-            Set<ExecutionAttemptID> groupedAttemptIds =
-                    executionAttemptsByLocation.getOrDefault(tmLocation, new HashSet<>());
+            for (AccessExecution execution : executionVertex.getCurrentExecutions()) {
```

Comment:
To double confirm, would any problem occur if one subtask has multiple attempts?

Reply:
The ExecutionAttemptID stands for a specific attempt of a specific execution. It seems the further processing doesn't care if there are ExecutionAttemptIDs belonging to the same execution. It processes each ExecutionAttemptID independently.

Reply:
I'm mainly not sure whether the web UI can correctly handle this.